pkg/cloudprovider/instances.go

```diff
@@ -30,7 +31,11 @@ func (k *k3s) InstanceExistsByProviderID(ctx context.Context, providerID string)
 }
 
 func (k *k3s) InstanceID(ctx context.Context, nodeName types.NodeName) (string, error) {
-	_, err := k.NodeCache.Get(string(nodeName))
+	if k.nodeInformerHasSynced == nil || !k.nodeInformerHasSynced() {
+		return "", errors.New("Node informer has not synced yet")
+	}
+
+	_, err := k.nodeInformer.Lister().Get(string(nodeName))
 	if err != nil {
 		return "", fmt.Errorf("Failed to find node %s: %v", nodeName, err)
 	}
```

Comment:
this impl seems weird from jump. we go through the trouble of listing the node and fail if it isnt there otherwise we return the nodeName that we passed? should we not be returning a stringified node uid or the like?

Reply:
i realize we are matching the legacy functionality here and changing would likely incur breakage

Reply:
Yeah I am not sure why we needed to do this. Honestly we could just implement InstancesV2() which gets all the info in one shot, but I wasn't going to tackle that here in a bugfix.